### DISCUSSION TITLE
LPS - CF Crash with example but work with pc.go_to command

### DISCUSSION THREAD

**Discussion by frulloa** [bitcraze/discussions#658] [Q&A]
Hello, please your kindly help.

**What do you think is the reason for the following behavior?** 
**The Crazyflie flies and follows good a position with one example but with another example crash all the time. I explain the issue in the following lines.**

- The Loco Position System (LPS) is set up as suggested on https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/ Furthermore, I tested it, and the position seems to be accurate. The LPS has 8 anchors and uses TDoA2.

We are testing examples provided by the Bitcraze team.

1.  The first example is https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/position_commander_demo.py on this example was added more points by pc.go_to command, see the following picture, and **the drone do its job** (youtube video: https://youtu.be/BswzMzoEWfQ)

![image](https://user-images.githubusercontent.com/78211872/231890444-598dab92-dac2-4fdd-bb58-52b8ec0e9faa.png)

2. Unfortunately, **the example autonomousSequence.py** https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/autonomousSequence.py **doesn't work well.** The Crazyflie starts well, but after a while, it crashes. Youtube video: https://youtu.be/jb_RB4Qfaq0 

- The Crazyflie client console shows this warning about Kalman. Maybe this has some relation with the improper behavior of autonomousSequence.py? I saw the code of that example, and it has some lines related to the Kalman filter.

![image](https://user-images.githubusercontent.com/78211872/231891604-54abc8a9-a559-4e67-9964-3f4b68e4f1d4.png)

**Comment by tobbeanton** [comment]
Debugging the position can be a bit difficult and judging from the video my guess is that there is an outlier that slipps though and causing the crash but could also be other things. But let's assume there is an outlier causing it.

1. The distance between the anchors might be a bit short as there are non-linearities close the the anchors that could cause problems. If you can increase the distance you could try that. Also try flying in TWR mode as this is a bit more robust and will most likely cause less outliers.

2. The M4 propeller or motor (shaft bent) is a bit unbalanced. Should work but exchanging it might be worth trying.

**Reply by frulloa** [reply]
Thanks for your response @tobbeanton I will consider your suggestions.
The distance between the anchors is 2.20[meters] in height x 2.25[m] in width, 4.10[m] in length. It is more than 2 meters, which is the suggested one. Anyway, I will try to increase the anchor's distance. 

But mainly, why do you consider work good if I use **pc.go_to** command? This command also employs LPS to have absolute position value, right? So, maybe that means that the LPS is correctly working

**Comment by krichardsson** [comment]
> The Crazyflie client console shows this warning about Kalman. Maybe this has some relation with the improper behavior of autonomousSequence.py? I saw the code of that example, and it has some lines related to the Kalman filter.

You don't have to worry about the kalman and stabilizer warnings, they are just caused by the propeller test.

The two scripts you are running, are slightly different in that the position_commander_demo uses the high level commander, while the autonomousSequence.py sends setpoints directly. In the end it should pretty much be the same result.
Have you tried to modify the coordinates to see if it makes any difference? Especially in the autonomousSequence script, moving all the points closer to move a shorter distance.
